client/setBase.mjs

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env node
+
+import path from "path";
+import fs from "fs";
+import process from "process";
+
+const filePath = path.join("public", "frame.html");
+
+// PureScript generates JS with the following import lines:
+//    `import * as Data_Foo from "../Data.Foo/index.js"
+// To remap `../Data.Foo/index.js` to `output/Data.Foo/index.js`
+// we append `/ignored/`.
+//
+// This:    `/output/ignored/../Data.Foo/index.js`
+// becomes: `/output/Data.Foo/index.js`
+const prodPath = "/output/ignored/"
+const devPath = "/js/output/ignored/"
+
+const environment = process.argv[2] || "dev";
+const baseHref = environment === "prod" ? prodPath : devPath;
+
+const frameHtml = fs.readFileSync(filePath, "utf-8");
+const newHtml = frameHtml
+  .split("\n")
+  .map((line) => line.replace(/^( *<base href=")[^"]*(".+)$/, `$1${baseHref}$2`))
+  .join("\n");
+
+fs.writeFileSync(filePath, newHtml);
```

Comment:
If we have to rewrite the file in place, I'm not sure this is the best way to do this. Will this dirty the source tree?

Reply:
I'm also curious how this works with the actual prod deployment of the UI. I'm not fully aware of all that goes into that.

Reply:
Yeah, it would dirty the source tree. 

The client is built in the `ci.yml` file by using NPM scripts before packaging up the results. When we deploy the code, the server downloads the `client.tar.gz`, unpacks it, and then runs it.

Reply:
How should this issue be resolved? Do we create 2 versions of the `frame.html` file, one with `base` set to one and another with `base` set to something else?

Reply:
I added a check in CI to fail the build if the `base` element does not refer to the production path. One can use `npm run base:dev` to test out their code in development and then run `npm run base:production` before committing the final results. This ensures the following:
- local dev still works without much friction in workflow
- the source tree is clean when the production build is made
- `frame.html` isn't duplicated, so there's only one place to update the import maps

I believe that resolves your final issue with this PR.